OpenNefia.Content/UI/Hud/HudUpdateSystem.cs

```diff
@@ -0,0 +1,33 @@
+﻿using OpenNefia.Content.World;
+using OpenNefia.Core.GameObjects;
+using OpenNefia.Core.IoC;
+using OpenNefia.Core.Maps;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.UI.Hud
+{
+    public sealed class HudUpdateSystem : EntitySystem
+    {
+        [Dependency] private readonly IHudLayer _hud = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        public override void Initialize()
+        {
+            SubscribeLocalEvent<MapOnHoursPassedEvent>(HandleMapHoursPassed, nameof(HandleMapHoursPassed));
+            SubscribeLocalEvent<ActiveMapChangedDelegate>(HandleMapChanged, nameof(HandleMapChanged));
+        }
+
+        private void HandleMapChanged(ref ActiveMapChangedDelegate ev)
+        {
+            _hud.UpdateMinimap();
```

Comment:
As mentioned before, if any time passes then the screen will always be refreshed when the player regains control. That means that any updates to the HUD before then won't be visible to the player. I think it would be better to have one `RefreshHud()` method in `IHudLayer` that's called when the screen is refreshed in `IFieldLayer`.

Reply:
changed in 9eb449f